pallets/core/src/modules/trust_registry/mod.rs

```diff
@@ -0,0 +1,360 @@
+    #[pallet::getter(fn registry_schema)]
+    pub type TrustRegistryStoredSchemas<T: Config> = StorageDoubleMap<
+        _,
+        Blake2_128Concat,
+        TrustRegistryId,
+        Blake2_128Concat,
+        TrustRegistrySchemaId,
+        (),
+    >;
+
+    /// Stores `TrustRegistry`s owned by conveners as a mapping of the form convener_id -> Set<registry_id>
+    #[pallet::storage]
+    #[pallet::getter(fn convener_trust_registries)]
+    pub type ConvenerTrustRegistries<T> =
+        StorageMap<_, Blake2_128Concat, Convener, TrustRegistryIdSet<T>, ValueQuery>;
+
+    /// Stores `Trust Registry`'s `Verifier`s schemas.
+    #[pallet::storage]
+    #[pallet::getter(fn trust_registry_verifier_schemas)]
+    pub type TrustRegistryVerifierSchemas<T: Config> = StorageDoubleMap<
+        _,
+        Blake2_128Concat,
+        TrustRegistryId,
+        Blake2_128Concat,
+        Verifier,
+        VerifierSchemas<T>,
+        ValueQuery,
+    >;
+
+    /// Stores `Trust Registry`'s `Issuer`s schemas.
+    #[pallet::storage]
+    #[pallet::getter(fn trust_registry_issuer_schemas)]
+    pub type TrustRegistryIssuerSchemas<T: Config> = StorageDoubleMap<
+        _,
+        Blake2_128Concat,
+        TrustRegistryId,
+        Blake2_128Concat,
+        Issuer,
+        IssuerSchemas<T>,
+        ValueQuery,
+    >;
+
+    /// Stores `Trust Registry`'s `Issuer`s configurations.
+    #[pallet::storage]
+    #[pallet::getter(fn trust_registry_issuer_config)]
+    pub type TrustRegistryIssuerConfigurations<T: Config> = StorageDoubleMap<
+        _,
+        Blake2_128Concat,
+        TrustRegistryId,
+        Blake2_128Concat,
+        Issuer,
+        TrustRegistryIssuerConfig<T>,
+        ValueQuery,
+    >;
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        /// Creates a new `Trust Registry` with the provided identifier.
+        /// The DID signature signer will be set as a `Trust Registry` owner.
+        #[pallet::weight(SubstrateWeight::<T>::init_trust_registry(init_trust_registry, sig))]
+        pub fn init_trust_registry(
+            origin: OriginFor<T>,
+            init_trust_registry: InitTrustRegistry<T>,
+            sig: DidOrDidMethodKeySignature<Convener>,
+        ) -> DispatchResult {
+            ensure_signed(origin)?;
+
+            WrappedActionWithNonce::new(
+                init_trust_registry.nonce(),
+                sig.signer(),
+                init_trust_registry,
+            )
+            .signed(sig)
+            .execute(
+                |WrappedActionWithNonce { action, .. }, registries, convener| {
+                    Self::init_trust_registry_(action, registries, convener)
+                },
+            )
+        }
+
+        /// Adds a new schema metadata entry (entries).
+        /// The DID signature signer must be the `Convener` owning this Trust Registry.
+        #[pallet::weight(SubstrateWeight::<T>::add_schema_metadata(add_schema_metadata, sig))]
+        pub fn add_schema_metadata(
```

Comment:
Is `add_schema_metadata` supposed to replace existing metadata? If yes, then it should be called `set_schema_metadata`

Reply:
It's not supposed